File: Android/Sample/lib.analytics/src/main/java/net/wequick/example/lib/analytics/Application.java
package net.wequick.example.lib.analytics;

import android.app.Activity;
import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;

import com.umeng.analytics.MobclickAgent;

/**
 * Created by galen on 16/5/26.
 */
public class Application extends android.app.Application {

    private class H extends Handler {
        @Override
        public void handleMessage(Message msg) {
            super.handleMessage(msg);
        }
    }

    private H mH;

    @Override
    public void onCreate() {
        super.onCreate();

        mH = new H();

        android.app.Application host = (android.app.Application) getApplicationContext();
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
            host.registerActivityLifecycleCallbacks(new ActivityLifecycleCallbacks() {
                @Override
                public void onActivityCreated(Activity activity, Bundle savedInstanceState) {

                }

                @Override
                public void onActivityStarted(Activity activity) {

                }

                @Override
                public void onActivityResumed(Activity activity) {
                    System.out.println("onActivityResumed " + activity.getClass().getName());
                    MobclickAgent.onResume(activity);
                }

                @Override
                public void onActivityPaused(Activity activity) {
                    System.out.println("onActivityPaused " + activity.getClass().getName());
                    MobclickAgent.onPause(activity);
                }

                @Override
                public void onActivityStopped(Activity activity) {

                }

                @Override
                public void onActivitySaveInstanceState(Activity activity, Bundle outState) {

                }

                @Override
                public void onActivityDestroyed(Activity activity) {

                }
            });
        }
    }
}

File: Android/Sample/lib.analytics/src/main/java/net/wequick/example/lib/analytics/Application.kt
package net.wequick.example.lib.analytics

import android.app.Activity
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.app.Application.ActivityLifecycleCallbacks

import com.umeng.analytics.MobclickAgent

/**
 * Created by galen on 16/5/26.
 */
class Application : android.app.Application() {

    private inner class H : Handler() {
        override fun handleMessage(msg: Message) {
            super.handleMessage(msg)
        }
    }

    private var mH: H? = null

    override fun onCreate() {
        super.onCreate()

        mH = H()

        val host = applicationContext as android.app.Application
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
            host.registerActivityLifecycleCallbacks(object : ActivityLifecycleCallbacks {
                override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {

                }

                override fun onActivityStarted(activity: Activity) {

                }

                override fun onActivityResumed(activity: Activity) {
                    System.out.println("onActivityResumed " + activity.javaClass.name)
                    MobclickAgent.onResume(activity)
                }

                override fun onActivityPaused(activity: Activity) {
                    System.out.println("onActivityPaused " + activity.javaClass.name)
                    MobclickAgent.onPause(activity)
                }

                override fun onActivityStopped(activity: Activity) {

                }

                override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle?) {

                }

                override fun onActivityDestroyed(activity: Activity) {

                }
            })
        }
    }
}
